frontend/src/components/ui/BuyMenu.jsx

```diff
@@ -1,19 +1,66 @@
+import { useState, useEffect } from "react";
 import { Text } from "@react-three/drei"
 import BuildGhostView from "../views/BuildGhostView";
 
-export default function BuyMenu({width, depth, towerConstructors, newTower, setNewTower}) {
+const NAME = "BuyMenu"
+
+export default function BuyMenu({game, towerConstructors}) {
+    const {gameMap, mouseInput} = game;
+    const {width, depth} = gameMap;
+
+    const [selected, setSelected] = useState(null);
```

Comment:
Looks like `selected` is not used. Is it different from `newTower`?

Reply:
Ah, I meant to save that for the next PR. I'll remove it.

I'm hoping to switch away from `newTower` so that we don't have to construct all of the duplicate towers. Also, to accommodate the new items for terraforming.